### ISSUE TITLE
[RFC] option to not focus input when closing date popper

### ISSUE THREAD

**Issue by benevbright** [mui/mui-x#8265]
### What's the problem? 🤔

DatePicker components set focus to input when the picker get closed.

https://github.com/mui/mui-x/blob/884ebc023dd1b553842e9b2babbbd58f7314c036/packages/x-date-pickers/src/internals/components/PickersPopper.tsx#L315

```
setTimeout(() => {
        if (lastFocusedElementRef.current instanceof HTMLElement) {
          lastFocusedElementRef.current.focus();
        }
      });
```

see the code above, it goes to the previous element that is set internally. `lastFocusedElementRef.current` is an internal ref in this component.

I like to have an option to not set focus to its own input field. 

Specifically it's a problem when I click another input field while the calendar picker is opened. When closing calendar by clicking another input, the calendar's own input takes the focus, not the one I click. 

The same thing happen with two `DatePicker` 😅.

Is there any solution?

**Comment by DanailH** [comment]
@alexfauquette should we close this one in favor of https://github.com/mui/mui-x/issues/6159?

**Comment by alexfauquette** [comment]
Closed in favor of #6159
